canvas_sdk/protocols/base.py

```diff
@@ -2,6 +2,7 @@
 
 
 class BaseProtocol:
-    def __init__(self, event) -> None:
+    def __init__(self, event, secrets = {}) -> None:
```

Comment:
this is a bug! you can't use a dict as a default argument because it will be shared among all instantiations of the class

see here: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Well good gracious golly that's a sharp stick. Thanks for catching and educating!

Reply:
it's the literal worst

Reply:
How's the updated version strike you?